elyra/pipeline/kfp/kfp_authentication.py

```diff
@@ -0,0 +1,605 @@
+                                              f'HTTP status code {resp.status_code}.',
+                                              provider=self._type,
+                                              request_history=request_history)
+                # set the login url
+                dex_login_url = resp.url
+
+            ################
+            # Attempt Dex Login
+            ################
+            resp = s.post(
+                dex_login_url,
+                data={"login": username, "password": password},
+                allow_redirects=True
+            )
+            request_history.append({'request_url': dex_login_url, 'response': resp})
+
+            if len(resp.history) == 0:
+                raise AuthenticationError('The LDAP credentials are probably invalid. '
+                                          f'Update runtime configuration \'{runtime_config_name}\' and try again.',
+                                          provider=self._type,
+                                          request_history=request_history)
+
+            # store the session cookies in a "key1=value1; key2=value2" string
+            return "; ".join([f"{c.name}={c.value}" for c in s.cookies])
+
+        # this code should never be reached; raise an error
+        raise AuthenticationError('An implementation problem was detected for LDAP authentication. '
+                                  'Please create an issue.',
+                                  provider=self._type,
+                                  request_history=request_history)
+
+
+class SATokenPathKFPAuthenticator(AbstractKFPAuthenticator):
+
+    _type = SupportedKFPAuthProviders.KF_PIPELINES_SA_TOKEN_PATH.value
+
+    def authenticate(self,
+                     kf_endpoint: str,
+                     runtime_config_name: str) -> Optional[str]:
+
+        request_history = []
+
+        # Verify the API endpoint
+        resp = requests.get(kf_endpoint)
+        request_history.append({'request_url': kf_endpoint, 'response': resp})
+        if resp.status_code != HTTPStatus.OK:
+            raise AuthenticationError(f'Error detecting whether Kubeflow server at {kf_endpoint} is secured: '
+                                      'HTTP status code {resp.status_code}'
+                                      f'Update runtime configuration \'{runtime_config_name}\' and try again.',
+                                      provider=self._type,
+                                      request_history=request_history)
+
+        # Running in a Kubernetes pod, kfp.Client can utilize environment
+        # variable KF_PIPELINES_SA_TOKEN_PATH for authentication. Verify
+        # that the variable is defined in the current environment.
+        env_var_name = 'KF_PIPELINES_SA_TOKEN_PATH'
+        if os.environ.get(env_var_name) is None:
+            raise AuthenticationError(f'Environment variable {env_var_name} is undefined.',
```

Comment:
There is no requirement to set `KF_PIPELINES_SA_TOKEN_PATH` when using this type of authentication.

If `KF_PIPELINES_SA_TOKEN_PATH` is unspecified, the `kfp.Client()` will use `/var/run/secrets/kubeflow/pipelines/token`.

Reply:
Fixed! This should now work as expected based on your earlier input.